src/main/java/de/tum/in/www1/artemis/service/connectors/LtiService.java

```diff
@@ -80,12 +81,15 @@
 
     private final LtiUserIdRepository ltiUserIdRepository;
 
+    private final HttpClient client;
+
     // Ok, Spring actually injects the response for the current context using a proxy
     // TODO Although this works, this is a bad design practice and we should move all response related code to the controller
     private final HttpServletResponse response;
 
     public final Map<String, Pair<LtiLaunchRequestDTO, Exercise>> launchRequestForSession = new HashMap<>();
 
+    @Autowired
```

Comment:
please don't use @Autowired here. We like to avoid it in production code

Reply:
we somehow need to tell spring which of those 2 contructors is our default contructor, since some classes refer to that

Reply:
still, we would like to avoid it, please remove the 2nd constructor!

Reply:
removed